bliss/simulator/prior.py

```diff
@@ -186,47 +193,52 @@ def _sample_star_fluxes(self, star_ratios):
-            for h, w, s, bnd in itertools.product(  # noqa: WPS352
-                range(self.n_tiles_h),
-                range(self.n_tiles_w),
-                range(self.max_sources),
-                range(self.max_bands),
-            ):
-                # sample a light source
-                flux_sample = obj_ratios[np.random.randint(0, len(obj_ratios) - 1)]
-                total_flux[b, h, w, s, bnd] = flux_sample[bnd] * flux_base[b, h, w, s]
+    def _load_color_models(self):
+        # Load models from disk
+        with open(f"{self.survey_data_dir}/color_models/star_gmm_nmgy.pkl", "rb") as f:
+            gmm_star = pickle.load(f)
+        with open(f"{self.survey_data_dir}/color_models/gal_gmm_nmgy.pkl", "rb") as f:
+            gmm_gal = pickle.load(f)
+        return gmm_star, gmm_gal
 
-        return total_flux
-
-    def _flux_ratios_against_b(self, survey_data_dir) -> Tuple[dict, dict]:
-        """Sample and compute all star, galaxy fluxes relative to `b`-band based on real image data.
+    def _flux_ratios(self) -> Tuple[Tensor, Tensor]:
+        """Sample and compute all star, galaxy fluxes based on real image data.
 
         Instead of pareto-sampling fluxes for each band, we pareto-sample `b`-band flux values,
         using prior `b`-band star/galaxy flux parameters, then apply other-band-to-`b`-band flux
         ratios to get other-band flux values. This is primarily because it is not always the case
         that, e.g.,
             flux_r ~ Pareto, flux_g ~ Pareto => flux_r | flux_g ~ Pareto.
-
-        Args:
-            survey_data_dir (str): path to survey data directory
+        This function is survey-specific as the 'b'-band index depends on the survey.
 
         Returns:
-            stars_fluxes (dict): image_id-indexed dict of star flux ratios # noqa: DAR202
-                relative to `b`-band
-            gals_fluxes (dict): image_id-indexed dict of galaxy flux ratios   # noqa: DAR202
-                relative to `b`-band
-
-        Raises:
-            NotImplementedError: if this method is not implemented in a subclass
+            stars_fluxes (Tensor): (b x th x tw x ms x nbands) Tensor containing all star flux
+                ratios for current batch
+            gals_fluxes (Tensor): (b x th x tw x ms x nbands) Tensor containing all gal fluxes
+                ratios for current batch
         """
-        raise NotImplementedError
+
+        samples = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
+        star_ratios_flat = self.gmm_star.sample(np.prod(samples))[0]
+        gal_ratios_flat = self.gmm_gal.sample(np.prod(samples))[0]
+
+        # Reshape drawn values into appropriate form
+        samples = samples + (self.n_bands - 1,)
+        star_ratios_nmgy = np.exp(np.reshape(star_ratios_flat, samples))
+        gal_ratios_nmgy = np.exp(np.reshape(gal_ratios_flat, samples))
+
+        hardcoded_ratios = np.array([219, 1039, 747, 139])
```

Comment:
What is this here for?

Reply:
forgot to delete that - it was a quick fix when I didn't have the `nmgy_to_nelec_dict` available.